rfcs/000-ml-feature-discovery.md

```diff
@@ -0,0 +1,183 @@
+
+# ML Features Discovery in Amundsen
+
+## Summary
+
+Amundsen currently indexes a few first-class entities, however we currently do not index any ML features that are easily discoverable for users. The goal of this work is to index ML features at the same level as Tables, Dashboards, and People resources in Amundsen.
+
+## Motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+Integration of ML projects in Amundsen has been heavily requested by the community and has already been implemented by other similar projects. This RFC outlines what will be the first step towards indexing ML projects in Amundsen since feature engineering is one of the most time consuming processes for ML engineers. Providing them with the ability to easily discover features that they can reuse for their models will significatly increase their productivity.
+
+## Guide-level Explanation (aka Product Details)
+
+> Explain the proposal as if it was already included in Amundsen and you were teaching it to an Amundsen user. That generally means:
+
+> Introducing new named concepts.
+> Explaining the feature largely in terms of examples.
+> Explaining how Amundsen users should think about the feature, and how it should impact the way they use Amundsen. It should explain the impact as concretely as possible.
+> If applicable, provide deprecation warnings, or migration guidance.
+> For implementation-oriented RFCs, this section should focus on how maintainers should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+- In the future this feature will hopefully replace all use of old Tag badges.
+
+### New Metadata Definitions
+#### Semantic Metadata
+Metadata related to the feature itself.
+- Key: what makes a feature uniquely identifiable is the group, name and version. We don’t use db/source because the same feature can exists in different places (like hive and dynamo)
+    - ex: test_feature_group_name/test_feature_name/1.2.0
+- Availability/sources: list of places where feature is available
+- Name: the name of the feature
+- Version: the version of the feature
+- Data type
+- Description: optional, a description of the feature
+- Owners: individual and team emails of owners
+- Feature group: a way of grouping features, can be defined differently in each company, in this case we assume this will be the name of the pipeline that produced the features
+- Entity: a way of grouping features semantically by using nouns
+- Owner tags: programmatically generated and not editable tags, useful to group features base don the specific engineering team using them
+- User tags: normal amundsen tags
+- Contains PII: weather a feature contains PII or not, will be a badge in the badges list
```

Comment:
Most of the features I've used in the past are some quantified value they are aggregated (e.g. avg daily balance, mean # of rides, log(daily sales), etc.) or they are transformed (e.g bins, categorical variables, etc.) where any form of PII data is removed as part of the calculation. Could you provide examples for how Contains PII would be used here?

Reply:
I think this varies depending on how you source your data, for us Amundsen simply indexes features from a different service at Lyft, so it is the responsibility of the feature creator to specify this.

Reply:
I agree that the creator should have the responsibility to specify if something is PII or not. In the same vein, why not just have the feature creator add a badge directly instead of having an attribute dedicated to PII?

Reply:
if you look at the common schema specified in the RFC there is no pii field, rather a badges list, so I completely agree.